spring-boot/proxy/src/main/java/eu/coatrack/proxy/security/ApiKeyValidityVerifier.java

```diff
@@ -0,0 +1,122 @@
+package eu.coatrack.proxy.security;
+
+/*-
+ * #%L
+ * coatrack-proxy
+ * %%
+ * Copyright (C) 2013 - 2021 Corizon | Institut für angewandte Systemtechnik Bremen GmbH (ATB)
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import eu.coatrack.api.ApiKey;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ *  This bean checks if an API key can be verified by considering the
+ *  the local API key list. It is used to evaluate the result of an
+ *  API key request received from admin.
+ *
+ *  @author Christoph Baier
+ */
+
+@Service
+public class ApiKeyValidityVerifier {
+
+    private static final Logger log = LoggerFactory.getLogger(ApiKeyValidityVerifier.class);
+
+    private final long oneHourInMillis = 1000 * 60 * 60;
+    private String apiKeyValue = "";
+    private List<String> apiKeyValueList = new ArrayList<>();
+    private Timestamp lastApiKeyValueListUpdate = new Timestamp(0);
+    private Timestamp adminsLocalTime = new Timestamp(0);
+
+    public boolean doesResultValidateApiKey(ResponseEntity<ApiKey> resultOfApiKeySearch, String apiKeyValue) {
+        if(apiKeyValue != null)
+            this.apiKeyValue = apiKeyValue;
+        else {
+            log.info("The API key value was null and could therefore not be checked for validity.");
```

Comment:
This should be log level `debug`. It is not required in normal production use.

Reply:
Does not exist anymore.